src/searchForm/autoComplete/overlay/AutoCompleteOverlay.tsx

```diff
@@ -14,24 +14,38 @@ export type AutoCompleteOverlayProps = Omit<
   readonly renderAutocompleteComponent: (
     props: Omit<AutoCompleteProps, 'searchForItems'>,
   ) => JSX.Element
+  readonly closeOnBlur: () => void
 }
 
 export const AutoCompleteOverlay = ({
   renderAutocompleteComponent,
+  closeOnBlur,
   ...props
 }: AutoCompleteOverlayProps) => {
+  const overlayRef = useRef(null)
```

Comment:
You can write it `useRef<HTMLDivElement>(null)` otherwise Typescript infers it as `React.MutableRefObject<any>`, and cast the `evt.relatedTarget as Node`